cli.js

```diff
@@ -22,40 +22,75 @@ var meow = require('meow');
 var path = require('path');
 var swPrecache = require('./');
 
-function setDefaults(cli) {
-  cli.flags.root = cli.flags.root || './';
-  if (cli.flags.root.lastIndexOf('/') !== cli.flags.root.length - 1) {
-    cli.flags.root += '/';
+function setDefaults(cli, configFileFlags) {
+  var compositeFlags = cli.flags;
```

Comment:
This could be easier to follow.

Perhaps something like:
```
// Merge command-line and config file flags
var flags = Object.assign(cli.flags, configFileFlags);
var defaults = {
  root: './',
  swFile: 'service-worker.js',
  cacheId: cli.pkg.name;
};

Object.assign(flags, defaults);

if (flags.root.lastIndexOf('/') !== flags.root.length - 1) {
  flags.root += '/';
}
flags.stripPrefix = flags.stripPrefix || flags.root;
flags.swFilePath = path.join(flags.root, flags.swFile);

(etc.)
```
(probably have to polyfill Object.assign if you still want to support older node).

Fewer characters to parse on each line is usually better for readability, and it's much easier to see what the default values actually are.